HIVE-22329 Create DriverContext

miklosgergely opened the pull request:
The current DriverContext class is not really a context, rather a queue, thus it should be renamed to DriverQueue. At the same time Driver has a a lot of fields, they should be in a real context class called DriverContext, which would also facilitate the extraction of independent parts of the Driver class.

kgyrtkirk commented on ql/src/java/org/apache/hadoop/hive/ql/DriverQueue.java:
```diff
@@ -0,0 +1,230 @@
+/**
+ * Handles the queue of tasks that should be executed by the driver.
+ */
+public class DriverQueue {
```
the apidoc describe it as "queue of tasks"; then I think it should be have the Task in it name...TaskQueue ?

miklosgergely replied:
fixed.

kgyrtkirk commented on ql/src/java/org/apache/hadoop/hive/ql/DriverQueue.java:
```diff
@@ -0,0 +1,230 @@
+  private static final int SLEEP_TIME = 2000;
+
+  private final Queue<Task<?>> runnable = new ConcurrentLinkedQueue<Task<?>>();;
+  private final Queue<TaskRunner> running = new LinkedBlockingQueue<TaskRunner>();
```
this seems to be mixed with the execution logic...the "queue" keyword looks less appropriate to me (but to be honest: no alternate idea)

miklosgergely replied:
fixed, you are right, it is not a queue

kgyrtkirk commented on ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLOperationContext.java:
```diff
@@ -34,19 +34,19 @@
   private final Hive db;
   private final HiveConf conf;
-  private final DriverContext driverContext;
+  private final DriverQueue driverQueue;
```
not sure about this; I think we might want to add this "queue" to the driverContext - as we have a lot of other stuff hanging around here...which could be moved into the context (conf?)

I would never think about asking for informations from a "queue"...(beyond the elements it might contain)

miklosgergely replied:
fixed.

kgyrtkirk commented on ql/src/java/org/apache/hadoop/hive/ql/ddl/table/lock/LockTableOperation.java:
```diff
@@ -34,7 +34,7 @@ public LockTableOperation(DDLOperationContext context, LockTableDesc desc) {
   @Override
   public int execute() throws HiveException {
-    Context ctx = context.getDriverContext().getCtx();
+    Context ctx = context.getDriverQueue().getCtx();
```
apparently this is not a queue: why would a queue would have `getCtx()` method

miklosgergely replied:
fixed.

miklosgergely commented:
@kgyrtkirk I've modified the patch according to your comments. Also modified the logic of passing the arguments to a test, now the TaskQueue and the context are passed at the initialize call, and execute has no argument, which is much cleaner. TaskQueue now doesn't exposes the context.
